src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +51,11 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     char buffer[4096];
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
+#if MSGHDR_REQUIRES_PADDING
+    msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, 0, NULL, 0, 0, 0};
+#else
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
+#endif
```

Comment:
The first two designators are `.msg_name` and `.msg_namelen`. Fixed.
Although the wrong members were assigned the values, all of the last CI jobs passed. Is this something untested in managed code?